Ginger/Unit Tests/NonUITests/GingerRunnerTests/GingerRunnerActionLogTest.cs

```diff
@@ -0,0 +1,283 @@
+
+            actDummy.EnableActionLogConfig = true;
+
+            //Act
+            gingerRunnerLogger.LogAction(actDummy);
+
+            //Assert
+            Assert.AreEqual(FindTextOccurrencesInFile(fileName, returnValExpected), 2);
+        }
+
+        [TestMethod]
+        public void TestActionLogInputAndReturnValues()
+        {
+            //Arrange
+            string fileName = TestResources.GetTempFile("ActionLog\\ActionLogTest_4.log");
+            string inputValExpected = "TestInputValue";
+            string returnValExpected = "123456";
+
+            GingerRunnerLogger gingerRunnerLogger = new GingerRunnerLogger(fileName);
+            ActDummy actDummy = new ActDummy();
+
+            // set all the values in the action
+            ActInputValue actInputValue = new ActInputValue();
+            actInputValue.ItemName = "TestInput";
+            actInputValue.Value = inputValExpected;
+
+            ActReturnValue actReturnValue = new ActReturnValue();
+            actReturnValue.ItemName = "TestReturn";
+            actReturnValue.Expected = returnValExpected;
+            actReturnValue.Actual = returnValExpected;
+
+            actDummy.InputValues.Add(actInputValue);
+            actDummy.ReturnValues.Add(actReturnValue);
+
+            actDummy.ActionLogConfig = new ActionLogConfig();
+            actDummy.ActionLogConfig.LogInputVariables = true;
+            actDummy.ActionLogConfig.LogOutputVariables = true;
+
+            actDummy.EnableActionLogConfig = true;
+
+            //Act
+            gingerRunnerLogger.LogAction(actDummy);
+
+            //Assert
+            Assert.IsTrue(IsFileContains(fileName, inputValExpected));
+            Assert.IsTrue(IsFileContains(fileName, returnValExpected));
+        }
+
+        [TestMethod]
+        public void TestActionLogStatusFailCheck()
+        {
+            //Arrange
+            string fileName = TestResources.GetTempFile("ActionLog\\ActionLogTest_5.log");
+            string returnValExpected = "123456";
+
+            GingerRunnerLogger gingerRunnerLogger = new GingerRunnerLogger(fileName);
+            ActDummy actDummy = new ActDummy();
+
+            ActReturnValue actReturnValue = new ActReturnValue();
+            actReturnValue.ItemName = "TestReturn";
+            actReturnValue.Expected = returnValExpected;
+            actReturnValue.Actual = string.Empty;
+
+            actDummy.ReturnValues.Add(actReturnValue);
+
+            actDummy.ActionLogConfig = new ActionLogConfig();
+            actDummy.ActionLogConfig.LogRunStatus = true;
+            actDummy.EnableActionLogConfig = true;
+
+            //actDummy.Execute();
+            actDummy.Status = Amdocs.Ginger.CoreNET.Execution.eRunStatus.Failed;
+
+            //Act
+            gingerRunnerLogger.LogAction(actDummy);
+
+            //Assert
+            Assert.IsTrue(IsFileContains(fileName, "Failed"));
+        }
+
+        [TestMethod]
+        public void TestActionLogStatusPassCheck()
```

Comment:
if this test is only for Pass value check remove all other settings, so test code will be minimal to verify only what is tested